ffi/capi/src/provider.rs

```diff
@@ -119,3 +120,19 @@ pub unsafe extern "C" fn icu4x_fs_data_provider_create(
         },
     }
 }
+
+#[no_mangle]
+/// Constructs an [`StaticDataProvider`] and retirns it as an [`ICU4XDataProvider`].
+/// See [`StaticDataProvider::new()`] for more details.
+///
+/// # Safety
+///
+/// Only access `provider` in the result if `success` is [`true`].
+pub unsafe extern "C" fn icu4x_static_data_provider_create() -> ICU4XCreateDataProviderResult {
```

Comment:
Thought: FWIW I would prefer it to be behind its own feature flag, since some non-WASM FFI users will likely want this too, but that's easy to change later.